test/e2e/framework/job/wait.go

```diff
@@ -133,3 +153,20 @@ func WaitForAllJobPodsGone(ctx context.Context, c clientset.Interface, ns, jobNa
 		return len(pods.Items) == 0, nil
 	})
 }
+
+// WaitForJobState waits for a job to be matched to the given condition.
+// The condition callback may use gomega.StopTrying to abort early.
+func WaitForJobState(ctx context.Context, c clientset.Interface, ns, jobName string, timeout time.Duration, state JobState) error {
+	return framework.Gomega().
+		Eventually(ctx, framework.RetryNotFound(framework.GetObject(c.BatchV1().Jobs(ns).Get, jobName, metav1.GetOptions{}))).
+		WithTimeout(timeout).
+		Should(framework.MakeMatcher(func(job *batchv1.Job) (func() string, error) {
+			matches := state(job)
+			if matches == "" {
+				return nil, nil
+			}
+			return func() string {
+				return fmt.Sprintf("%v\n%s", matches, format.Object(job, 1))
```

Comment:
```suggestion
				return fmt.Sprintf("%v:\n%s", matches, format.Object(job, 1))
```

Reply:
My definition of "short string" was a bit too vague. It might also be a short sentence, including a dot at the end. Then adding the `:` would look weird - we would have to explicitly say that the returned string is meant to be a prefix.

Let's keep it as is.

/approve